src/udp_wrap.cc

```diff
@@ -445,7 +445,16 @@ void UDPWrap::OnRecv(uv_udp_t* handle,
     return;
   }
 
-  char* base = static_cast<char*>(realloc(buf->base, nread));
+  char* base = NULL;
+  if (nread != 0) {
+     base = static_cast<char*>(realloc(buf->base, nread));
+    // If realloc fails, fallback to old pointer
+    // safe since we are always shrinking the buffer
+    if(base == NULL)
+        base = buf->base;
+  } else
+    free(buff->base);
```

Comment:
Also forgot `buf->base=NULL` for second branch. I think it is more correct to write `buf->base = base` after both branches. I.e.:

```
if (nread) {
     base = static_cast<char*>(realloc(buf->base, nread));
     if (base == NULL)
          base = buf->base;
     else
          buf->base = base;
} else {
    free(buf->base);
    buf->base = base = NULL;
}

```